google-cloud-firestore/src/test/java/com/google/cloud/firestore/LocalFirestoreHelper.java

```diff
@@ -889,4 +928,72 @@ public static String autoId() {
   public static Map<String, Object> fromSingleQuotedString(String json) {
     return fromJsonString(json.replace("'", "\""));
   }
+
+  static class RequestResponseMap
```

Comment:
I feel this should no longer be called `RequestResponseMap`, this does much more now. Can you come up with a more specific name?

I also think we could split this into two classes - the base RequestResponseMap and a subclass that enforces the one active request setting. With good naming (tm) it would now be super obvious what is going on.